pyteal/ast/app.py

```diff
@@ -356,3 +356,56 @@ def address(cls, app: Expr) -> MaybeValue:
 
 
 AppParam.__module__ = "pyteal"
+
+
+class AppParamObject:
+    """Represents information about an application's parameters"""
+
+    def __init__(self, app: Expr) -> None:
+        """Create a new AppParamObject for the given application.
+
+        Args:
+            app: An identifier for the app. It must be an index into Txn.ForeignApps that
+                corresponds to the app to check, or since v4, an application ID that appears in
+                Txn.ForeignApps or is the CurrentApplicationID. In either case, it must evaluate to
+                uint64.
+        """
+        self._app: Final = app
+
+    def approval_program(self) -> MaybeValue:
+        """Get the bytecode of Approval Program for the application."""
+        return AppParam.approvalProgram(self._app)
+
+    def clear_state_program(self) -> MaybeValue:
+        return AppParam.clearStateProgram(self._app)
+
+    def global_num_unit(self) -> MaybeValue:
+        """Get the number of uint64 values allowed in Global State for the application."""
+        return AppParam.globalNumUnit(self._app)
+
+    def global_num_byte_slice(self) -> MaybeValue:
+        """Get the number of byte array values allowed in Global State for the application."""
+        return AppParam.globalNumByteSlice(self._app)
+
+    def local_num_unit(self) -> MaybeValue:
```

Comment:
```suggestion
    def global_num_uint(self) -> MaybeValue:
        """Get the number of uint64 values allowed in Global State for the application."""
        return AppParam.globalNumUnit(self._app)

    def global_num_byte_slice(self) -> MaybeValue:
        """Get the number of byte array values allowed in Global State for the application."""
        return AppParam.globalNumByteSlice(self._app)

    def local_num_uint(self) -> MaybeValue:
```

Unfortunately, these method names are misspelled in the `AppParam` class, but there's not need to perpetuate the type in a brand new class.

I'm suggesting `uint` (which aligns with **algod**) as opposed to `int` (which aligns with **goal**)